### PR TITLE
PLENIGO-759 & PLENIGO-764

### PR DESCRIPTION
**Pull request by SebaSOFT**
Includes solutions of:

PLENIGO-764 - Deactive metered for single articles
PLENIGO-759 - Sign up an product in wordpress redirects to main page

### CODE REVIEW

File: plenigo_plugin/PlenigoContentManager.php [hunk 1]
```diff
@@ -151,22 +153,35 @@ public function add_scripts() {
     public function plenigo_js_snippet() {
         PlenigoSDKManager::get()->getPlenigoSDK();
         $isPaywalled = $this->plenigo_paywalled_content();
-        /*
-          echo '<script type="application/javascript">'
-          . 'var plenigo = plenigo || {};'
-          . 'plenigo.baseURI = "' . PLENIGO_SVC_URL . '";'
-          . 'plenigo.baseStaticURI = "' . PLENIGO_SVC_URL . '";</script>'; */
 
+        if (DEF_PLENIGO_SVC_URL != PLENIGO_SVC_URL) {
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoContentManager.php]: What is when PLENIGO_SVC_URL is null or empty, or cant happen this?

File: plenigo_plugin/PlenigoContentManager.php [hunk 2]
```diff
@@ -65,17 +65,19 @@ class PlenigoContentManager {
      * (for shortcode processing) but more than 1.
      */
     const PLENIGO_CONTENT_PRIO = 5;
-    //Plenigo settings group
+//Plenigo settings group
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoContentManager.php]: Why you moved all the comments to the left?

File: plenigo_plugin/PlenigoContentManager.php [hunk 3]
```diff
@@ -533,8 +557,13 @@ private function plenigo_check() {
 
             // If the user hasn't actually bought the product, check for free views
             if ($res === FALSE) {
-                plenigo_log_message("USER DIDNT BOUGHT THE PRODUCT, CHECKING FOR FREE VIEWS");
-                $res = PlenigoSDKManager::get()->plenigo_has_free_views();
+                if ($tagExempt === FALSE) {
+                    plenigo_log_message("USER DIDNT BOUGHT THE PRODUCT, CHECKING FOR FREE VIEWS");
+                    $res = PlenigoSDKManager::get()->plenigo_has_free_views();
+                }else{
```
Comment (robertholfeld) [inline plenigo_plugin/PlenigoContentManager.php]: Space between } else {

File: plenigo_plugin/PlenigoContentManager.php [hunk 4]
```diff
@@ -234,8 +249,8 @@ private function plenigo_filter_content($content, $isFeed = FALSE) {
 
                 $curtain_code = $content;
 
-                //IF the blog is configured to show only excerpts in the RSS, then we let the content
-                //go if paywalled and MORE tag not found, otherwise we don't show anything
+//IF the blog is configured to show only excerpts in the RSS, then we let the content
```
Comment (maximilianschweitzer) [inline plenigo_plugin/PlenigoContentManager.php]: IF == If

File: plenigo_plugin/PlenigoContentManager.php [hunk 5]
```diff
@@ -520,6 +535,15 @@ private function plenigo_check() {
         plenigo_log_message("LOOKING FOR THE FULL PLENIGO SDK CHECK");
         $this->addDebugLine("Plenigo backend check");
         $sdk = PlenigoSDKManager::get()->getPlenigoSDK();
+        //Checking if the metered view is exempt by tag
+        $tagExempt = FALSE;
+        $optExempt = isset($this->options[self::OPT_METERED_EXEMPT]) ? $this->options[self::OPT_METERED_EXEMPT] : NULL;
```
Comment (maximilianschweitzer) [inline plenigo_plugin/PlenigoContentManager.php]: Duplicated code with line 174

File: plenigo_plugin/PlenigoLoginManager.php [hunk 6]
```diff
@@ -340,8 +354,23 @@ public function update_with_plenigo_user($id, $userData) {
      * This method stores the last URL inside the site regardless the HTTP referrer
      */
     public function store_url() {
+        plenigo_log_message("PREVIOUS THROWBACK: " . var_export($_SESSION['plenigo_throwback_url'], true), E_USER_NOTICE);
+
+
         $current_url = PlenigoURLManager::get()->getSanitizedURL();
-        $_SESSION['plenigo_throwback_url'] = $current_url;
+        $arrTokens = explode(',', self::PLENIGO_URL_EXCEPTIONS);
+        $booUpdate = true;
```
Comment (maximilianschweitzer) [inline plenigo_plugin/PlenigoLoginManager.php]: booUpdate?

File: readme.txt [hunk 7]
```diff
@@ -3,7 +3,7 @@ Contributors: Sebastian Dieguez <s.dieguez@plenigo.com>
 Tags: paywall, e-commerce, Ecommerce, paid content software, subscriptions, newspaper, media, pay-per-read, pay, plugin, donate, money, transaction, bank, visa, mastercard, credit, debit, card, widget, give, pay what you want, plenigo, payment
 Requires at least: 4.0.0
 Tested up to: 4.1.1
-Stable tag: 1.1.32
+Stable tag: 1.2.1
```
Comment (maximilianschweitzer) [inline readme.txt]: We don't have a 1.2.0. Perhaps we should start with that :-D

File: plenigo_wordpress.php [hunk 8]
```diff
@@ -39,8 +39,12 @@
 
 // Plenigo JavaScript SDK / Services
 define('PLENIGO_SVC_URL', "https://api.plenigo.com");
+define('PLENIGO_OAUTH_SVC_URL', "https://www.plenigo.com");
 define('PLENIGO_JSSDK_URL', "https://static.plenigo.com");
 
+// Default URL for Javascript
+define('DEF_PLENIGO_SVC_URL', "https://api.plenigo.com");
```
Comment (maximilianschweitzer) [inline plenigo_wordpress.php]: Default URL for JS? Which JS?

File: plenigo_sdk/plenigo/internal/server-interface/payment/Checkout.php [hunk 9]
```diff
@@ -243,18 +242,33 @@ public function setTestMode($ts)
...
+     */
+    public function setCurrency($currency) {
+        $this->currency = $currency;
+    }
+
+    /**
+     * Sets the Title.
+     *
+     * @param string $title The title.
+     */public function setTitle($title) {
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/internal/server-interface/payment/Checkout.php]: line break

File: plenigo_sdk/plenigo/internal/server-interface/payment/Checkout.php [hunk 10]
```diff
@@ -274,4 +288,14 @@ public function getMap()
         return $map;
     }
 
+    private function performValidation() {
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/internal/server-interface/payment/Checkout.php:295]: Comment missing

File: plenigo_sdk/plenigo/internal/services/Service.php [hunk 11]
```diff
@@ -61,13 +68,20 @@ protected static function getRequest($endPoint, array $params = array())
...
+     * @param bool $oauth TRUE if the needed request is going to the OAuth API.
      * @param array  $params   Optional params to pass to the request.
      *
      * @return the request result.
      */
-    protected static function postRequest($endPoint, array $params)
+    protected static function postRequest($endPoint, $oauth = false, array $params = array())
     {
-        $url = PlenigoManager::get()->getUrl() . $endPoint;
+        if($oauth){
```
Comment (robertholfeld) [inline plenigo_sdk/plenigo/internal/services/Service.php:78]: Looks like the same code like in getRequest(..) only with different notice.

File: readme.txt [hunk 12]
```diff
@@ -129,6 +129,14 @@ separator, native Wordpress tag.
 Coming soon
 
 == Changelog ==
+= 1.2.1 - A new hope =
+- Feature: Now you can set a post tag to be a "metered views prevention tag"
+- Improved: Updated PHP SDK
+- Improved: Moved metered settings to the new "metered views" section
+- Fixed: A problem with returning URLs after loging in from the curtain
+- Fixed: A problem where not being loggedout from plenigo when logging out from Wordpress via the login screen
```
Comment (robertholfeld) [inline readme.txt]: loggedout

File: plenigo_plugin/PlenigoContentManager.php (hunk 1 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoContentManager.php]: We are talking about global constants, The main reason for this if is to have code run if the endpoint is our test endpoint.

File: plenigo_plugin/PlenigoContentManager.php (hunk 2 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoContentManager.php]: formatter made amistake

File: plenigo_plugin/PlenigoContentManager.php (hunk 5 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoContentManager.php]: Done

File: plenigo_plugin/PlenigoLoginManager.php (hunk 6 above)
Comment (SebaSOFT) [inline plenigo_plugin/PlenigoLoginManager.php]: I had a flashback to my Microsoft naing conventions for variable names

File: readme.txt (hunk 7 above)
Comment (SebaSOFT) [inline readme.txt]: Also we are WP 4.2 compatible

File: plenigo_wordpress.php (hunk 8 above)
Comment (SebaSOFT) [inline plenigo_wordpress.php]: This is for comparing and detecting If I changed the JavaScript SDk origin to our test system and automatically put that into plenigo.baseURI

File: plenigo_sdk/plenigo/internal/server-interface/payment/Checkout.php (hunk 9 above)
Comment (SebaSOFT) [inline plenigo_sdk/plenigo/internal/server-interface/payment/Checkout.php]: Will correct it for the next SDK update

File: plenigo_sdk/plenigo/internal/server-interface/payment/Checkout.php (hunk 10 above)
Comment (SebaSOFT) [inline plenigo_sdk/plenigo/internal/server-interface/payment/Checkout.php:295]: Will correct it for the next SDK update

**Pull request merged**